src/main/java/org/padaiyal/utilities/commandline/CommandLineUtility.java

```diff
@@ -0,0 +1,447 @@
+              timeOutDuration.getSeconds()
+          )
+      );
+    }
+    HashMap<StdType, String> output = new HashMap<>();
+    final int returnCode = process.waitFor();
+    final Instant executionEndTimestamp = Instant.now();
+    output.put(StdType.STDOUT,
+        StreamUtility.convertInputStreamToString(process.getInputStream()));
+    output.put(StdType.STDERR,
+        StreamUtility.convertInputStreamToString(process.getErrorStream()));
+    process.destroy();
+    return new Response(returnCode, output, executionStartTimestamp, executionEndTimestamp);
+  }
+
+  /**
+   * Executes the specified command.
+   *
+   * @param command               Command to execute.
+   * @param commandLine           Type of command line to use to execute the command.
+   * @param timeOutDuration       Time out for the command execution.
+   * @return                      The response of the command.
+   * @throws IOException          Thrown if there is an issue executing the command.
+   * @throws InterruptedException Thrown if the execution of the command is interrupted.
+   * @throws TimeoutException     Thrown if the command execution exceeds specified timeout.
+   */
+  public static synchronized Response executeCommand(
+      String command,
+      CommandLine commandLine,
+      Duration timeOutDuration
+  ) throws IOException, InterruptedException, TimeoutException {
+    Objects.requireNonNull(
+        command,
+        I18nUtility.getFormattedString(
+            "CommandLineUtility.input.validation.nonNull",
+            "Command"
+        )
+    );
+    Objects.requireNonNull(
+        timeOutDuration,
+        I18nUtility.getFormattedString(
+            "CommandLineUtility.input.validation.nonNull",
+            "Duration"
+        )
+    );
+
+    String[] splitCommand;
+    splitCommand = new String[]{
+        // Path to command line
+        commandLines.get(commandLine)
+            .toAbsolutePath()
+            .toString(),
+        // Command line switch
+        CommandLine.getCommandLineSwitch(commandLine),
+        // Command
+        command.replaceAll("\"", "\\\"")
+      };
+
+    return executeCommand(splitCommand, timeOutDuration);
+  }
+
+  /**
+   * Executes the specified command.
+   *
+   * @param command                       Command to execute.
+   * @param timeOutDuration               Time out for the command execution.
+   * @return                              The response of the command.
+   * @throws IOException                  Thrown if there is an issue executing the command.
+   * @throws InterruptedException         Thrown if the execution of the command is interrupted.
+   * @throws TimeoutException             Thrown if the command execution exceeds specified timeout.
+   * @throws CommandLineNotFoundException Thrown if the command line expected for the 
+   *                                      operating system is not available.
+   */
+  public static Response executeCommand(
```

Comment:
Explain the difference in behaviour between this executeCommand vs the other ones.

Reply:
Not needed. The arguments make it evident.